spec/requests/players_spec.rb

```diff
@@ -0,0 +1,139 @@
+require "rails_helper"
+require "./spec/support/request_helper"
+
+RSpec.describe Player, type: :request do
+  include RequestSpecHelper
+
+  let!(:team) { create(:team) }
+  let!(:player) { create_list(:player, 10, team_id: team.id) }
+
+  let(:team_id) { team.id }
+  let(:player_id) { player.first.id }
+
+  let(:player_params) {
+    {
+        first_name: "James",
+        second_name: "Kariuki",
+        last_name: "Kaguna",
+        nick_name: "jimnah",
+        id_number: "31262778",
+        dob: "1997-09-19",
+        phone_number: "0715739940",
+        team_id: team_id
+    }
+  }
```

Comment:
Did you try using **FactoryBot** methods?
- it has an `attributes_for` method which achieves what this block is doing. 
ie 
```ruby
...
let(:player_params) { attributes_for(player.first) }
...
```